src/porting_android.cpp

```diff
@@ -64,85 +63,73 @@ void android_main(android_app *app)
 	{
-		errorstream << "Java_net_minetest_minetest_GameActivity_putMessageBoxResult got: "
-				<< std::string((const char*)env->GetStringChars(text,0))
-				<< std::endl;
+		errorstream <<
+			"Java_net_minetest_minetest_GameActivity_putMessageBoxResult got: " <<
+			std::string((const char*) env->GetStringChars(text, nullptr)) << std::endl;
 	}
 }
 
 namespace porting {
-
-std::string path_storage = DIR_DELIM "sdcard" DIR_DELIM;
-
-android_app* app_global;
-JNIEnv*      jnienv;
+android_app *app_global;
+JNIEnv      *jnienv;
 jclass       nativeActivity;
 
-jclass findClass(std::string classname)
+jclass findClass(const std::string &classname)
 {
-	if (jnienv == 0) {
-		return 0;
-	}
+	if (jnienv == nullptr)
+		return nullptr;
 
 	jclass nativeactivity = jnienv->FindClass("android/app/NativeActivity");
-	jmethodID getClassLoader =
-			jnienv->GetMethodID(nativeactivity,"getClassLoader",
-					"()Ljava/lang/ClassLoader;");
-	jobject cls =
-			jnienv->CallObjectMethod(app_global->activity->clazz, getClassLoader);
+	jmethodID getClassLoader = jnienv->GetMethodID(
+			nativeactivity, "getClassLoader", "()Ljava/lang/ClassLoader;");
+	jobject cls = jnienv->CallObjectMethod(
+						app_global->activity->clazz, getClassLoader);
 	jclass classLoader = jnienv->FindClass("java/lang/ClassLoader");
-	jmethodID findClass =
-			jnienv->GetMethodID(classLoader, "loadClass",
+	jmethodID findClass = jnienv->GetMethodID(classLoader, "loadClass",
 					"(Ljava/lang/String;)Ljava/lang/Class;");
-	jstring strClassName =
-			jnienv->NewStringUTF(classname.c_str());
+	jstring strClassName = jnienv->NewStringUTF(classname.c_str());
 	return (jclass) jnienv->CallObjectMethod(cls, findClass, strClassName);
 }
 
 void initAndroid()
 {
-	porting::jnienv = NULL;
+	porting::jnienv = nullptr;
 	JavaVM *jvm = app_global->activity->vm;
 	JavaVMAttachArgs lJavaVMAttachArgs;
 	lJavaVMAttachArgs.version = JNI_VERSION_1_6;
 	lJavaVMAttachArgs.name = PROJECT_NAME_C "NativeThread";
-	lJavaVMAttachArgs.group = NULL;
-#ifdef NDEBUG
-	// This is a ugly hack as arm v7a non debuggable builds crash without this
-	// printf ... if someone finds out why please fix it!
-	infostream << "Attaching native thread. " << std::endl;
-#endif
-	if ( jvm->AttachCurrentThread(&porting::jnienv, &lJavaVMAttachArgs) == JNI_ERR) {
+	lJavaVMAttachArgs.group = nullptr;
+
+	if (jvm->AttachCurrentThread(&porting::jnienv, &lJavaVMAttachArgs) == JNI_ERR) {
 		errorstream << "Failed to attach native thread to jvm" << std::endl;
 		exit(-1);
 	}
 
 	nativeActivity = findClass("net/minetest/minetest/GameActivity");
-	if (nativeActivity == 0) {
+	if (nativeActivity == nullptr)
 		errorstream <<
 			"porting::initAndroid unable to find java native activity class" <<
 			std::endl;
-	}
 
 #ifdef GPROF
-	/* in the start-up code */
+	// in the start-up code
 	__android_log_print(ANDROID_LOG_ERROR, PROJECT_NAME_C,
 			"Initializing GPROF profiler");
-	monstartup("libminetest.so");
+	monstartup("libMinetest.so");
```

Comment:
why upper case here ? lib are lower case in unix in 99.9% of the cases

Reply:
name changed in my previous PR

Reply:
ah